wazo_dird/database/queries/base.py

```diff
@@ -87,3 +101,52 @@ def _get_dird_user(self, session, user_uuid):
             session.flush()
 
         return user
+
+    def _populate_parameters(self, parameters=None):
+        new_params = dict(self.DEFAULTS)
+        if parameters:
+            parameters.setdefault('offset', self.DEFAULTS['offset'])
+            new_params.update(parameters)
+
+        return new_params
+
+    def _apply_search_params(self, rows, order, limit, offset, reverse):
+        if order:
+            try:
+                rows = sorted(
+                    rows,
+                    key=lambda x: unicodedata.normalize('NFKD', x[order]),
+                    reverse=reverse,
+                )
+            except KeyError:
+                raise ValueError(f"order: column '{order}' was not found")
+        elif reverse:
+            rows.reverse()
+
+        if not limit:
+            return rows[offset:]
+        else:
+            return rows[offset : offset + limit]
```

Comment:
since we are returning a slice anyway, I would avoid mutating the `rows` parameter(or mutating parameters in general), which might cause issues if the list passed to this method is otherwise reused.
https://en.wikipedia.org/wiki/Referential_transparency

instead of `rows.reverse`, `rows = reversed(rows)` would produces an iterator, and `itertools.islice` would slice any iterable or iterator and should minimize copying.

```
if order:
    rows = ...
elif reversed:
    rows = reversed(rows)

return list(itertools.islice(rows, offset, offset + limit if limit else None))
```
here reversed does no copying, and a new list is generated by the `list` call.